### DISCUSSION TITLE
How to calcutae the receiving power of a packet?

### DISCUSSION THREAD

**Discussion by Ambica2022** [inet-framework/inet#931] [General]
Hi,

I am unable to locate the code related to "calculation of receivng power expressed in dBm of each packet".
Any clue is appreciated.Thanks in advance.

Regards,
Ambica

**Comment by n-jay** [comment]
Hi @Ambica2022,
from my understanding you can get the RSSI value using the `packetlevel/SignalTag_m.h` file as follows:

```
#include "inet/physicallayer/wireless/common/contract/packetlevel/SignalTag_m.h"

if (receivedPkt->findTag<SignalPowerInd>() != nullptr) {
	  auto signalPowerInd = receivedPkt->getTag<SignalPowerInd>();
	  double rxPower = signalPowerInd->getPower().get();
	  EV_INFO << "RX power= " << rxPower << "W" << endl;
}
```

Depending on your version of INET, the location of the `SignalTag_m.h` can vary, this should however work on the latest.
